test/apps/iodemo/io_demo.cc

```diff
@@ -697,17 +740,28 @@ class DemoServer : public P2pDemoCommon {
         iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
         w->init(this, conn, msg->sn, iov, &_curr_state.write_count);
 
+        std::map<uint32_t, conn_state_t>::iterator it =
+                _conn_stat_map.find(conn->id());
+        assert(it != _conn_stat_map.end());
+        it->second.write_add(msg->data_size);
         recv_data(conn, *iov, msg->sn, w);
     }
 
     virtual void dispatch_connection_accepted(UcxConnection* conn) {
+        bool res = _conn_stat_map.insert(std::make_pair(conn->id(),
```

Comment:
maybe _conn_stat_map[conn->id] = conn_state_t() ?

Reply:
that's the benefit? actually `insert` has one comparison less vs `operator[]` (we expect "insert unique" path) and ability to assert result